fdbserver/Ratekeeper.actor.cpp

```diff
@@ -367,17 +396,27 @@ class RatekeeperImpl {
 
 	ACTOR static Future<Void> refreshStorageServerCommitCosts(Ratekeeper* self) {
 		state double lastBusiestCommitTagPick;
+		state std::vector<Future<Void>> replies;
 		loop {
 			lastBusiestCommitTagPick = now();
 			wait(delay(SERVER_KNOBS->TAG_MEASUREMENT_INTERVAL));
+
+			replies.clear();
+
 			double elapsed = now() - lastBusiestCommitTagPick;
 			// for each SS, select the busiest commit tag from ssTrTagCommitCost
 			for (auto& [ssId, ssQueueInfo] : self->storageQueueInfo) {
-				ssQueueInfo.refreshCommitCost(elapsed);
+				// NOTE: In some cases, for unknown reason SS will not respond to the updateCommitCostRequest. Since the
+				// information is not time-sensitive, place a timeout to avoid stucking RK.
```

Comment:
This probably shouldn't be blocking RK at all. If we want to make it a separate message like this rather than piggybacking on the existing one, you could potentially just skip the wait below. Hold them all in the vector and the `TAG_MEASUREMENT_INTERVAL` delay happening above will let them keep running for that duration, after which they will get cleared away and cancelled anyway.

Reply:
This makes sense. A timeout seems unnecessary in this case.